volmdlr/faces.py

```diff
@@ -5242,6 +5428,246 @@ def triangulation_lines(self, resolution=50):
                                       volmdlr.Point2D(v_max, v)))
         return lines_x, lines_y
 
+    
+    def pair_with(self, other_bspline_face3d):
```

Comment:
About the refactoring of BSplineFace3D.pair_with, please detach `def extremities`, `def adjacent_direction` and `def corresponding_directions_grids` from `def pair_with`. They can be individual methods for BSplineFace3D, using _self_.

Reply:
It's done!